.github/workflows/build.yml

```diff
@@ -53,15 +54,15 @@ jobs:
         dotnet paket install
 
     - name: Cache SonarCloud packages
-      uses: actions/cache@v1
+      uses: actions/cache@v4
       with:
         path: ~/sonar/cache
         key: ${{ runner.os }}-sonar
         restore-keys: ${{ runner.os }}-sonar
         
     - name: Cache SonarCloud scanner
       id: cache-sonar-scanner
-      uses: actions/cache@v1
+      uses: actions/cache@v4
```

Comment:
Also change restore packages to
        dotnet tool restore
        dotnet paket restore
        dotnet restore